timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -1512,25 +1512,43 @@ void declare_fanout_summed_potential(py::module &m) {
         .def("get_potentials", &FanoutSummedPotential::get_potentials);
 }
 
-template <typename RealType> void declare_log_sum_exp(py::module &m, const char *typestr) {
+template <typename RealType> void declare_segmented_sum_exp(py::module &m, const char *typestr) {
 
-    using Class = LogSumExp<RealType>;
-    std::string pyclass_name = std::string("LogSumExp_") + typestr;
+    using Class = SegmentedSumExp<RealType>;
+    std::string pyclass_name = std::string("SegmentedSumExp_") + typestr;
     py::class_<Class, std::shared_ptr<Class>>(m, pyclass_name.c_str(), py::buffer_protocol(), py::dynamic_attr())
-        .def(py::init([](const int N) { return new Class(N); }), py::arg("N"))
         .def(
-            "sum",
-            [](Class &summer,
-               const py::array_t<double, py::array::c_style> &values) -> py::array_t<RealType, py::array::c_style> {
-                std::vector<RealType> h_vals = py_array_to_vector_with_cast<double, RealType>(values);
-                int N = h_vals.size();
-                py::array_t<RealType, py::array::c_style> py_res(1);
-                summer.sum_host(N, &h_vals[0], py_res.mutable_data());
-
-                return py_res;
+            py::init([](const int max_vals_per_segment, const int segments) {
+                return new Class(max_vals_per_segment, segments);
+            }),
+            py::arg("max_vals_per_segment"),
+            py::arg("num_segments"))
+        .def(
+            "logsumexp",
+            [](Class &summer, const std::vector<std::vector<double>> &vals) -> std::vector<RealType> {
```

Comment:
nit: this results in a Python binding that accepts lists; would it be preferable to accept arrays instead (e.g. like potentials)

Reply:
This is intentional to be able to provide ragged arrays to test the segment logic. In practice the code will typically be working on ragged arrays (as defined by segments, still a single buffer).